optuna_dashboard/ts/components/GraphRank.tsx

```diff
@@ -0,0 +1,342 @@
+import * as plotly from "plotly.js-dist-min"
+import React, { FC, useEffect, useState } from "react"
+import {
+  Grid,
+  FormControl,
+  FormLabel,
+  MenuItem,
+  Select,
+  Typography,
+  SelectChangeEvent,
+  useTheme,
+  Box,
+} from "@mui/material"
+import { plotlyDarkTemplate } from "./PlotlyDarkMode"
+import { makeHovertext } from "../graphUtil"
+import { useMergedUnionSearchSpace } from "../searchSpace"
+
+const PADDING_RATIO = 0.05
+const plotDomId = "graph-rank"
+
+interface AxisInfo {
+  name: string
+  range: [number, number]
+  is_log: boolean
+  is_cat: boolean
+}
+
+interface RankPlotInfo {
+  xaxis: AxisInfo
+  yaxis: AxisInfo
+  xvalues: (string | number)[]
+  yvalues: (string | number)[]
+  zvalues: number[]
+  colors: number[]
+  hovertext: string[]
+}
+
+export const GraphRank: FC<{
+  study: StudyDetail | null
+}> = ({ study = null }) => {
+  const theme = useTheme()
+  const [objectiveId, setobjectiveId] = useState<number>(0)
+  const searchSpace = useMergedUnionSearchSpace(study?.union_search_space)
+  const [xParam, setXParam] = useState<SearchSpaceItem | null>(null)
+  const [yParam, setYParam] = useState<SearchSpaceItem | null>(null)
+  const objectiveNames: string[] = study?.objective_names || []
+
+  if (xParam == null && searchSpace.length > 0) {
```

Comment:
[question] Do you intentionally use `==` and `===` everywhere?

Reply:
Thank you for your comment. I don't have a specific policy, but I think we should standardize one way or the other. I will use `===`.